packages/core/src/model.js

```diff
@@ -988,6 +1013,40 @@ ${associationOwner._getAssociationDebugList()}`);
       await this.queryInterface.addIndex(tableName, index, options);
     }
 
+    const existingConstraints = await this.queryInterface.showConstraints(tableName, {
+      ...options,
+      constraintType: 'FOREIGN KEY',
+    });
+
+    const associations = Object.values(this.modelDefinition.associations)
+      .filter(association => {
+        return association.options?.foreignKey?.keys?.length > 0;
+      })
+      .filter(association => {
+        return association.associationType !== 'HasOne';
```

Comment:
'Hack' to be able to create keys from associations

Reply:
there is a 'topological' sort that still needs fixing, but this is a quick hack that works for now, long term we need to fix the topo sort to handle composite keys with HasOne